### PR TITLE
Add README on how to generate docs

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made
Took a bit to figure this out so thought best to document it

Did some clean up of the CONTRIBUTION.md file while at it:
- Move the installing of the virtual env above testing.
- Renamed it from "Using your local version" to "Developing and Running SQLFluff Locally"
- Made titles in Title case consistently.
- Indented all the Nerdy Details under Nerdy Details.

### Are there any other side effects of this change that we should be aware of?
Nope.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/parser` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.
- - Update a couple of references from `master` to `main`.

### CODE REVIEW

**Comment by tunetheweb** [comment]
FYI @zhongjiajie

File: CONTRIBUTING.md
Comment (zhongjiajie) [inline CONTRIBUTING.md:85]:
```suggestion

> `python3 -m venv .venv` creates a new virtual environment (in current working
```

File: docs/README.md
Comment (zhongjiajie) [inline docs/README.md:49]:
```suggestion
The config is available in `docs/source/conf.py`.
```

File: docs/README.md
Comment (zhongjiajie) [inline docs/README.md:26]:
```suggestion
Again, this command is run from the root server, not the `docs` subfolder but you
```

File: docs/README.md
Comment (zhongjiajie) [inline docs/README.md:15]:
```suggestion
make -C docs clean
```

File: docs/README.md
Comment (zhongjiajie) [inline docs/README.md:12]:
```suggestion
root directory (drop the `-C docs` if running from within the `docs` directory).
```

**Review by zhongjiajie — COMMENTED** [review]
Add some minor suggestions, BTW, good write 👍

**Comment by tunetheweb** [comment]
> Add some minor suggestions

Good spots! Thanks.

@sqlfluff/sqlfluff-maintainers think this is good to merge now if one of you can approve?

**Review by pwildenhain — APPROVED** [review]
Great work! Let's merge

**Pull request merged**
